infrastructure/io/src/test/java/tech/pegasys/teku/infrastructure/io/resource/FileResourceLoaderTest.java

```diff
@@ -43,10 +43,7 @@ public void shouldReturnEmptyWhenFileDoesNotExist(@TempDir Path tempDir) throws
   }
 
   @Test
-  public void shouldCreateInputStreamWhenPathIsADirectory(@TempDir Path tempDir) throws Exception {
-    // We could potentially return empty for directories, but it is going to be confusing for users
-    // to say we couldn't find something that exists. We should report that we couldn't read it.
-    assertThat(loader.load(tempDir.toAbsolutePath().toString())).isNotEmpty();
```

Comment:
@ajsutton when calling `Files.newInputStream(path)` with directory `path` on Windows there is the following exception: 
```
java.nio.file.AccessDeniedException: C:\Users\Admin\AppData\Local\Temp\junit10004047955021046840

	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:370)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:421)
	at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)
	at java.base/java.nio.file.Files.newInputStream(Files.java:155)
	at tech.pegasys.teku.infrastructure.io.resource.FileResourceLoader.load(FileResourceLoader.java:30)
	at tech.pegasys.teku.infrastructure.io.resource.FileResourceLoaderTest.shouldCreateInputStreamWhenPathIsADirectory(FileResourceLoaderTest.java:49)
```
Not sure I got the rationale behind your comment and what `InputStream` should be returned in case of directory.
I would suggest to either return `empty` when the `path` is directory or throw an exception.

Reply:
You'll get an IOException of some form on all systems trying to actually read a directory.  The point of this behaviour is that this method returns an empty optional if the file couldn't be found, it throws an `IOException` if it can't be read.  Since directories do exist but can't be read, the expected behaviour is to throw `IOException` (or any subclass).  Both ways fail, but this way gives a much more accurate error message to the user.

Reply:
Ah I see, on Windows it throws earlier than on UNIX.  Throwing is still the right answer rather than returning empty, just need to update the test to handle the exception being thrown either from load or from the read call.